owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/status/StatusRequestor.kt

```diff
@@ -0,0 +1,113 @@
+package com.owncloud.android.lib.resources.status
+
+import com.owncloud.android.lib.common.OwnCloudClient
+import com.owncloud.android.lib.common.http.HttpConstants
+import com.owncloud.android.lib.common.http.methods.nonwebdav.GetMethod
+import com.owncloud.android.lib.common.operations.RemoteOperationResult
+import org.json.JSONObject
+import java.net.URL
+import java.util.concurrent.TimeUnit
+
+internal class StatusRequestor {
+
+    companion object {
+        /**
+         * Maximum time to wait for a response from the server when the connection is being tested,
+         * in MILLISECONDs.
+         */
+        private const val TRY_CONNECTION_TIMEOUT: Long = 5000
+        private const val NODE_INSTALLED = "installed"
+        private const val HTTPS_SCHEME = "https"
+        private const val HTTP_SCHEME = "http"
+        private const val NODE_VERSION = "version"
+    }
+
+    private fun checkIfConnectionIsRedirectedToNoneSecure(
+        isConnectionSecure: Boolean,
+        baseUrl: String,
+        redirectedUrl: String
+    ): Boolean {
+        return isConnectionSecure ||
+                (baseUrl.startsWith(HTTPS_SCHEME) && redirectedUrl.startsWith(
+                    HTTP_SCHEME
+                ))
+    }
+
+    fun updateLocationWithRedirectPath(oldLocation: String, redirectedLocation: String): String {
+        if (!redirectedLocation.startsWith("/"))
+            return redirectedLocation
+        val oldLocation = URL(oldLocation)
+        return URL(oldLocation.protocol, oldLocation.host, oldLocation.port, redirectedLocation).toString()
+    }
+
+    private fun getGetMethod(url: String): GetMethod {
+        return GetMethod(URL(url + OwnCloudClient.STATUS_PATH)).apply {
+            setReadTimeout(TRY_CONNECTION_TIMEOUT, TimeUnit.SECONDS)
+            setConnectionTimeout(TRY_CONNECTION_TIMEOUT, TimeUnit.SECONDS)
+        }
+    }
+
+    data class RequestResult(
+        val getMethod: GetMethod,
+        val status: Int,
+        val result: RemoteOperationResult<OwnCloudVersion>,
+        val redirectedToUnsecureLocation: Boolean
+    )
```

Comment:
What is the purpose of StatusRequestor and RequestResult? Do you think we could use them somewhere else? Should we try to make them global? Could we use them on each remote operation(Obviously not in this PR)?

Reply:
No but it reduces the size of GetRemoteStatusOperation

Reply:
Ok, LGTM for the moment. We could take advantage of this RequestResult for future refactors 👍